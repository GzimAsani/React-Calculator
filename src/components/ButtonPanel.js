import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

function ButtonPanel({ clickHandler }) {
  return (
    <div className="main-div">
      <div className="group-1">
        <Button clickHandler={clickHandler} name="AC" />
        <Button name="+/-" clickHandler={clickHandler} />
        <Button name="%" clickHandler={clickHandler} />
        <Button name="รท" clickHandler={clickHandler} />
      </div>
      <div className="group-2">
        <Button name="7" clickHandler={clickHandler} />
        <Button name="8" clickHandler={clickHandler} />
        <Button name="9" clickHandler={clickHandler} />
        <Button name="x" clickHandler={clickHandler} />
      </div>
      <div className="group-3">
        <Button name="4" clickHandler={clickHandler} />
        <Button name="5" clickHandler={clickHandler} />
        <Button name="6" clickHandler={clickHandler} />
        <Button name="-" clickHandler={clickHandler} />
      </div>
      <div className="group-4">
        <Button name="1" clickHandler={clickHandler} />
        <Button name="2" clickHandler={clickHandler} />
        <Button name="3" clickHandler={clickHandler} />
        <Button name="+" clickHandler={clickHandler} />
      </div>
      <div className="group-5">
        <Button name="0" clickHandler={clickHandler} />
        <Button name="." clickHandler={clickHandler} />
        <Button name="=" clickHandler={clickHandler} />
      </div>
    </div>
  );
}

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func,
};

ButtonPanel.defaultProps = {
  clickHandler: null,
};

export default ButtonPanel;
