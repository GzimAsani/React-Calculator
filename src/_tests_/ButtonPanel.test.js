import { create } from 'react-test-renderer';
import ButtonPanel from '../components/ButtonPanel';

test('render correctly', () => {
  const handleClick = () => { ''; };
  const tree = create(<ButtonPanel clickHandler={handleClick} name="" />);
  expect(tree).toMatchSnapshot();
});
