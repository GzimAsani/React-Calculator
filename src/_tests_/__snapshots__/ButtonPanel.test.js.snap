// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="main-div"
>
  <div
    className="group-1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="group-2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="group-3"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="group-4"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="group-5"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
